display_list/display_list_color_source.h

```diff
@@ -0,0 +1,668 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_DISPLAY_LIST_DISPLAY_LIST_COLOR_SOURCE_H_
+#define FLUTTER_DISPLAY_LIST_DISPLAY_LIST_COLOR_SOURCE_H_
+
+#include "flutter/display_list/display_list.h"
+#include "flutter/display_list/display_list_attributes.h"
+#include "flutter/display_list/types.h"
+#include "flutter/fml/logging.h"
+#include "third_party/skia/include/core/SkShader.h"
+#include "third_party/skia/include/effects/SkGradientShader.h"
+
+namespace flutter {
+
+class DlColorColorSource;
+class DlImageColorSource;
+class DlLinearGradientColorSource;
+class DlRadialGradientColorSource;
+class DlConicalGradientColorSource;
+class DlSweepGradientColorSource;
+class DlUnknownColorSource;
+
+enum class DlColorSourceType {
+  kColor,
+  kImage,
+  kLinearGradient,
+  kRadialGradient,
+  kConicalGradient,
+  kSweepGradient,
+  kUnknown
+};
+
+enum class DlTileMode {
+  // Replicate the edge color if the shader draws outside of its
```

Comment:
In comment documentation here and elsewhere, you refer to shaders but the enum is specified on the color source. It may have been written before you introduced the concept of color sources in display lists.

Reply:
Yes, and/or I just cut and pasted from SkTileMode.

Will do.

Reply:
Done.